tests/test_driver.py

```diff
@@ -185,37 +185,69 @@ def test_get_outputs(self, driver, carol_pubkey,
             'output_index': 0
         } in outputs
 
-    @mark.parametrize('spent,outputs_qty', ((False, 1), (True, 1), (None, 2)))
-    def test_get_outputs_with_spent_query_param(
-            self, spent, outputs_qty, driver, carol_pubkey,
-            persisted_carol_bicycle_transaction,
-            persisted_carol_car_transaction,
-            persisted_transfer_carol_car_to_dimi):
-        outputs = driver.outputs.get(carol_pubkey, spent=spent)
-        assert len(outputs) == outputs_qty
-
-        # Return only unspent outputs
-        if spent is False:
-            assert {
-                'transaction_id': persisted_carol_bicycle_transaction['id'],
-                'output_index': 0
-            } in outputs
-        # Return only spent outputs
-        elif spent is True:
-            assert {
-                'transaction_id': persisted_carol_car_transaction['id'],
-                'output_index': 0
-            } in outputs
-        # Return all outputs for carol
-        elif spent is None:
-            assert {
-                'transaction_id': persisted_carol_bicycle_transaction['id'],
-                'output_index': 0
-            } in outputs
-            assert {
-                'transaction_id': persisted_carol_car_transaction['id'],
-                'output_index': 0
-            } in outputs
+    def test_get_outputs_with_spent_query_param(self, driver):
+        from bigchaindb_driver.crypto import generate_keypair
+        import uuid
+
+        test_parameters = [(False, 2, 1), (True, 0, 1), (None, 2, 2)]
+
+        def create_transaction():
+            return driver.transactions.prepare(
+                    operation='CREATE',
+                    signers=carol.public_key,
+                    asset={
+                        'data': {
+                            'asset': {
+                                'serial_number': str(uuid.uuid4()),
+                                'manufacturer': str(uuid.uuid4()),
+                            },
+                        },
+                    },
+            )
+
+        carol, dimi = generate_keypair(), generate_keypair()
+        # create the first transaction for carol
+        create_tx1 = create_transaction()
+        create_tx1 = driver.transactions.fulfill(
+            create_tx1, private_keys=carol.private_key)
+        driver.transactions.send_commit(create_tx1)
+        # create the second transaction for carol
+        create_tx2 = create_transaction()
+        create_tx2 = driver.transactions.fulfill(
+            create_tx2, private_keys=carol.private_key)
+        driver.transactions.send_commit(create_tx2)
+        for entry in test_parameters:
```

Comment:
I think it's easier to read if you just unwrap this loop and write the three `assert`s. You can also remove `test_parameters` and just use the values directly.